### PR TITLE
feat: PX-3912 scroll to the partner header on tab change

### PR DESCRIPTION
**Pull request by kiryl-zubarau**
This PR ignores default scroll behavior for the partner profile page and adds scroll to the partner header on tab change.

JIRA - https://artsyproduct.atlassian.net/browse/PX-3912

ezgif com-gif-maker (13)

### CODE REVIEW

File: src/v2/Components/NavBar/useNavBarHeigth.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { themeProps } from "@artsy/palette"
+import { useMatchMedia } from "v2/Utils/Hooks/useMatchMedia"
+import { MOBILE_NAV_HEIGHT, NAV_BAR_HEIGHT } from "./constants"
+
+export const useNavBarHeigth = (
+  mobileMediaQueryString: string = themeProps.mediaQueries.xs
+) => {
+  const isMobile = useMatchMedia(mobileMediaQueryString)
```
Comment (damassi) [inline src/v2/Components/NavBar/useNavBarHeigth.ts:8]: We'll want to be sure that this works during the server-side render pass and with JS turned off

Comment (kiryl-zubarau) [reply src/v2/Components/NavBar/useNavBarHeigth.ts:8]: @damassi is there a way how we identify a user's device at the server-side render time?

`useMatchMedia` doesn't work during the server-side render
https://github.com/artsy/force/blob/89d35f7144aafe4987967f9ac774783bfa2c2374/src/v2/Utils/Hooks/useMatchMedia.ts#L40-L43

File: src/v2/Apps/Partner/Components/NavigationTabs/index.tsx [hunk 2]
```diff
@@ -15,6 +17,7 @@ interface NavigationTabsProps {
 }
 
 export const NavigationTabs: React.FC<NavigationTabsProps> = ({ partner }) => {
+  const navBarHeigth = useNavBarHeigth()
```
[the remark this answers is not in the document]
Comment (kiryl-zubarau) [reply src/v2/Apps/Partner/Components/NavigationTabs/index.tsx]: Fixed

File: src/typings/sharify.d.ts [hunk 3]
```diff
@@ -57,6 +57,7 @@ declare module "sharify" {
       readonly BRAZE_API_KEY: string
       readonly BRAZE_API_URL: string
       readonly BRAZE_LOGGING: boolean
+      readonly INCLUDE_ESCAPED_FRAGMENT: boolean
```
[the remark this answers is not in the document]
Comment (kiryl-zubarau) [reply src/typings/sharify.d.ts]: This from the meta info branch, missed it when switched between branches. Removed

File: src/v2/Components/NavBar/useNavBarHeigth.ts (hunk 1 above)
Comment (damassi) [reply src/v2/Components/NavBar/useNavBarHeigth.ts]: I'm not entirely sure about the context around this bit of code, but the way we genenerally do server-side compatable layouts is via our library fresnel: https://github.com/artsy/fresnel. 

There's quite a few examples of its use throughtout our codebase; just search for `<Media at` or `<Media greaterThan=`. With those pieces one can do some clever things like:

```
<Media at='sm'>{renderLayout({ isMobilie: true })}</Media>
```

Not the prettiest, but it works for when one needs an `isMobile`-ish variable.

File: src/v2/Apps/Partner/Routes/Artists/index.tsx [hunk 4]
```diff
@@ -23,13 +22,10 @@ export const ArtistsRoute: React.FC<ArtistsRouteProps> = ({
...
 }) => {
-  const isMobile = useMatchMedia(themeProps.mediaQueries.xs)
+  const navBarHeight = useNavBarHeight()
 
   const handleArtistClick = () => {
-    const offset =
-      PARTHER_NAV_BAR_HEIGHT +
-      (isMobile ? MOBILE_NAV_HEIGHT : NAV_BAR_HEIGHT) +
-      20
+    const offset = PARTHER_NAV_BAR_HEIGHT + navBarHeight + 20
```
Comment (dzucconi) [inline src/v2/Apps/Partner/Routes/Artists/index.tsx:28]: Note the typo in the const `PARTHER_NAV_BAR_HEIGHT`

Comment (kiryl-zubarau) [reply src/v2/Apps/Partner/Routes/Artists/index.tsx]: Fixed

File: src/v2/Components/NavBar/useNavBarHeigth.ts (hunk 1 above)
Comment (kiryl-zubarau) [reply src/v2/Components/NavBar/useNavBarHeigth.ts]: I rewrote the components to use `Media` component, please take a look

File: src/v2/Apps/Partner/Components/NavigationTabs/index.tsx [hunk 5]
```diff
@@ -80,7 +89,12 @@ export const NavigationTabs: React.FC<NavigationTabsProps> = ({ partner }) => {
             style={stuck ? { boxShadow: DROP_SHADOW } : undefined}
           >
             <HorizontalPadding maxWidth={breakpoints.xl}>
-              <RouteTabs fill>{renderTabs()}</RouteTabs>
+              <Media greaterThan="xs">
+                <RouteTabs fill>{renderTabs(NAV_BAR_HEIGHT)}</RouteTabs>
+              </Media>
+              <Media at="xs">
+                <RouteTabs fill>{renderTabs(MOBILE_NAV_HEIGHT)}</RouteTabs>
+              </Media>
```
Comment (damassi) [inline src/v2/Apps/Partner/Components/NavigationTabs/index.tsx:97]: Perfect 👍

**Comment by sweir27** [comment]
Needs a rebase, otherwise 👍 from me.

**Pull request merged**